core/services/relay/evm/evmtesting/chain_reader_interface_tester.go

```diff
@@ -0,0 +1,330 @@
+	return []clcommontypes.BoundContract{
+		{Name: AnyContractName, Address: it.address, Pending: true},
+		{Name: AnySecondContractName, Address: it.address2, Pending: true},
+	}
+}
+
+type testStructFn = func(*chain_reader_tester.ChainReaderTesterTransactor, *bind.TransactOpts, int32, string, uint8, [32]uint8, common.Address, []common.Address, *big.Int, chain_reader_tester.MidLevelTestStruct) (*evmtypes.Transaction, error)
+
+func (it *EvmChainReaderInterfaceTester[T]) sendTxWithTestStruct(t T, testStruct *TestStruct, fn testStructFn) {
+	tx, err := fn(
+		&it.evmTest.ChainReaderTesterTransactor,
+		it.GetAuthWithGasSet(t),
+		*testStruct.Field,
+		testStruct.DifferentField,
+		uint8(testStruct.OracleID),
+		ConvertOracleIDs(testStruct.OracleIDs),
+		common.Address(testStruct.Account),
+		ConvertAccounts(testStruct.Accounts),
+		testStruct.BigField,
+		MidToInternalType(testStruct.NestedStruct),
+	)
+	require.NoError(t, err)
+	it.Helper.Commit()
+	it.IncNonce()
+	it.AwaitTx(t, tx)
+	it.dirtyContracts = true
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) GetAuthWithGasSet(t T) *bind.TransactOpts {
+	gasPrice, err := it.client.SuggestGasPrice(it.Helper.Context(t))
+	require.NoError(t, err)
+	extra := new(big.Int).Mul(gasPrice, big.NewInt(it.Helper.GasPriceBufferPercent()))
+	extra = extra.Div(extra, big.NewInt(100))
+	it.auth.GasPrice = gasPrice.Add(gasPrice, extra)
+	return it.auth
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) IncNonce() {
+	if it.auth.Nonce == nil {
+		it.auth.Nonce = big.NewInt(1)
+	} else {
+		it.auth.Nonce = it.auth.Nonce.Add(it.auth.Nonce, big.NewInt(1))
+	}
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) AwaitTx(t T, tx *evmtypes.Transaction) {
+	ctx := it.Helper.Context(t)
+	receipt, err := bind.WaitMined(ctx, it.client, tx)
+	require.NoError(t, err)
+	require.Equal(t, evmtypes.ReceiptStatusSuccessful, receipt.Status)
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) deployNewContracts(t T) {
+	// First test deploy both contracts, otherwise only deploy contracts if cleanup decides that we need to.
+	if it.address == "" {
+		it.address = it.deployNewContract(t)
+		it.address2 = it.deployNewContract(t)
+	} else if it.evmTest == nil {
+		it.address = it.deployNewContract(t)
+		it.dirtyContracts = false
+	}
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) deployNewContract(t T) string {
+	// 105528 was in the error: gas too low: have 0, want 105528
+	// Not sure if there's a better way to get it.
+	it.auth.GasLimit = 10552800
+
+	address, tx, ts, err := chain_reader_tester.DeployChainReaderTester(it.GetAuthWithGasSet(t), it.Helper.Backend())
+	require.NoError(t, err)
+	it.Helper.Commit()
+	if it.evmTest == nil {
+		it.evmTest = ts
+	}
+
+	it.IncNonce()
+	it.AwaitTx(t, tx)
+	return address.String()
+}
+
+func (it *EvmChainReaderInterfaceTester[T]) MaxWaitTimeForEvents() time.Duration {
+	return it.Helper.MaxWaitTimeForEvents()
+}
+
+func ConvertOracleIDs(oracleIDs [32]commontypes.OracleID) [32]byte {
```

Comment:
Maybe consider a type so you can call a method here too:
```go
OracleIDs(oracleIDs).Convert()
```
Seems like more could be said than convert too, but I'm not sure what.

Reply:
OracleIdsToBytes?